src/rez/rezconfig.py

```diff
@@ -1080,8 +1113,15 @@
 # * :data:`rez_1_environment_variables` will be set to False.
 # * :data:`rez_1_cmake_variables` will be set to False.
 #
-# You should aim to do this - it will mean your packages are more strictly
+# You should aim to do this. It will mean your packages are more strictly
 # validated, and you can more easily use future versions of Rez.
+#
+# .. versionchanged:: 2.114.0
+#    Now enabled by default.
+#
+# .. deprecated:: 2.114.0
+#    Will be removed in a future release. Additionally, the default will change
+#    from disabled to enabled in rez 3.0.0.
 disable_rez_1_compatibility = False
```

Comment:
I initially wanted to deprecate it and remove it in 3.0.0, but I found out that this would be a little bit too disruptive for our oldest users. Same for `rez_1_environment_variables`. `rez_1_environment_variables` is even more painful because it's a lot harder to know if packages are actively using these old environment variables or not.